homeassistant/components/tankerkoenig/__init__.py

```diff
@@ -69,87 +79,87 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
 
-    latitude = conf.get(CONF_LATITUDE, hass.config.latitude)
-    longitude = conf.get(CONF_LONGITUDE, hass.config.longitude)
-    radius = conf[CONF_RADIUS]
-    additional_stations = conf[CONF_STATIONS]
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set a tankerkoenig configuration entry up."""
+    tankerkoenig = TankerkoenigData(hass, entry)
 
-    setup_ok = await hass.async_add_executor_job(
-        tankerkoenig.setup, latitude, longitude, radius, additional_stations
-    )
+    setup_ok = await hass.async_add_executor_job(tankerkoenig.setup)
     if not setup_ok:
         _LOGGER.error("Could not setup integration")
         return False
 
-    hass.data[DOMAIN] = tankerkoenig
-
-    hass.async_create_task(
-        async_load_platform(
-            hass,
-            SENSOR_DOMAIN,
-            DOMAIN,
-            discovered=tankerkoenig.stations,
-            hass_config=conf,
-        )
+    tankerkoenig.undo_update_listener = entry.add_update_listener(
+        _async_update_listener
     )
 
+    hass.data.setdefault(DOMAIN, {})
+    hass.data[DOMAIN][entry.unique_id] = tankerkoenig
+
+    hass.config_entries.async_setup_platforms(entry, [Platform.SENSOR])
+
     return True
 
 
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Unload Tankerkoenig config entry."""
+    unload_ok = await hass.config_entries.async_unload_platforms(
+        entry, [Platform.SENSOR]
+    )
+    if unload_ok:
+        entry_data: TankerkoenigData = hass.data[DOMAIN][entry.unique_id]
+        assert entry_data.undo_update_listener is not None
+        entry_data.undo_update_listener()
+        hass.data[DOMAIN].pop(entry.unique_id)
+
+    return unload_ok
+
+
+async def _async_update_listener(hass: HomeAssistant, entry: ConfigEntry) -> None:
+    """Handle options update."""
+    await hass.config_entries.async_reload(entry.entry_id)
+
+
 class TankerkoenigData:
     """Get the latest data from the API."""
 
-    def __init__(self, hass, conf):
+    def __init__(self, hass: HomeAssistant, entry: ConfigEntry) -> None:
         """Initialize the data object."""
-        self._api_key = conf[CONF_API_KEY]
-        self.stations = {}
-        self.fuel_types = conf[CONF_FUEL_TYPES]
-        self.update_interval = conf[CONF_SCAN_INTERVAL]
-        self.show_on_map = conf[CONF_SHOW_ON_MAP]
+        self._api_key = entry.data[CONF_API_KEY]
+        self._selected_stations = entry.data[CONF_STATIONS]
         self._hass = hass
+        self.undo_update_listener: CALLBACK_TYPE | None = None
+        self.stations: dict[str, dict] = {}
+        self.fuel_types = entry.data[CONF_FUEL_TYPES]
+        self.show_on_map = entry.options[CONF_SHOW_ON_MAP]
 
-    def setup(self, latitude, longitude, radius, additional_stations):
+    def setup(self):
         """Set up the tankerkoenig API.
 
         Read the initial data from the server, to initialize the list of fuel stations to monitor.
```

Comment:
You might want to update the description, since now the data is not initialized here but in the config flow

Reply:
good point 👍